pkg/oc/bootstrap/docker/openshift/servicecatalog.go

```diff
@@ -98,6 +101,44 @@ func (h *Helper) InstallServiceCatalog(f *clientcmd.Factory, configDir, publicMa
 		return errors.NewError("cannot instantiate service catalog template").WithCause(err)
 	}
 
+	// once the TSB is created, find the service account it created for the SC to use when calling
+	// the TSB and look up the token associated w/ the SA.
+	var token []byte
+	err = wait.Poll(1*time.Second, 1*time.Minute, func() (bool, error) {
+		glog.V(2).Info("polling for service catalog service account secrets")
+		sa, err := kubeClient.Core().ServiceAccounts(tsbNamespace).Get("servicecatalog-client", metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		for _, s := range sa.Secrets {
+			if strings.HasPrefix(s.Name, "servicecatalog-client-token") {
+				secret, err := kubeClient.Core().Secrets(tsbNamespace).Get(s.Name, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				if secret.Annotations["kubernetes.io/created-by"] == "openshift.io/create-dockercfg-secrets" {
+					continue
+				}
+				token = secret.Data["token"]
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+	if err != nil || len(token) == 0 {
+		return errors.NewError(fmt.Sprintf("failed to locate the service catalog client token: %v", err))
+	}
+
+	// create a new secret containing the SA's token, in the catalog's namespace.
+	// when we register the TSB w/ the SC, we'll reference this new secret.
```

Comment:
@deads2k @pmorie i wasn't sure if i should do this (create a new secret in the SC's namespace w/ the TSB SA's token) or just point the SC to the TSB SA's existing secret in the TSB's namespace (and then grant the SC access to view secrets in the TSB's namespace).  This seems like a better answer, but either direction has the opportunity for escalation attacks.